src/XIVLauncher.Common.Unix/Compatibility/WineRunner.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Serilog;
+using XIVLauncher.Common.Util;
+
+namespace XIVLauncher.Common.Unix.Compatibility;
+
+public class WineRunner
+{
+    public string RunCommand { get; }
+
+    public string RunArguments { get; }
+
+    public string WineServer { get; }
+
+    public string PathArguments { get; }
+
+    private string wineFolder { get; }
+
+    private string downloadUrl;
+
+    private string toolFolder;
+
+    private string xlcoreFolder;
+
+    public Dictionary<string, string> Environment { get; }
+
+    public WineRunner(string runCmd, string runArgs, string folder, string url, string rootFolder, Dictionary<string, string> env = null, bool is64only = false)
+    {
+        RunArguments = runArgs;
+        wineFolder = folder;
+        downloadUrl = url;
+        toolFolder = Path.Combine(rootFolder, "compatibilitytool", "beta");
+        xlcoreFolder = rootFolder;
+        Environment = (env is null) ? new Dictionary<string, string>() : env;
+        if (string.IsNullOrEmpty(runCmd))
+        {
+            RunCommand = Path.Combine(toolFolder, folder, "bin", (is64only) ? "wine" : "wine64");
+            WineServer = Path.Combine(toolFolder, folder, "bin", "wineserver");
+        }
+        else
+        {
+            if (File.Exists(Path.Combine(runCmd, "wine64")))
+                RunCommand = Path.Combine(runCmd, "wine64");
+            else if (File.Exists(Path.Combine(runCmd, "wine")))
+                RunCommand = Path.Combine(runCmd, "wine");
+            else
+                throw new FileNotFoundException($"There is no wine or wine64 binary at {runCmd}.");
+            WineServer = Path.Combine(runCmd, "wineserver");
+        }
+    }
+
+    public async Task Install()
+    {
+        if (CompatibilityTools.IsDirectoryEmpty(Path.Combine(toolFolder, wineFolder)))
```

Comment:
The wine "runner" should have a concept of it is pointing to custom user provided wine or is internally managed and guard against that here.